### PR TITLE
PL adapter refactoring

### PR DESCRIPTION
**Pull request by mihran113**
Refactor `AimLogger` class for pytorch lightning to follow the guidelines for PL loggers and provide more flexibility for the users to derive and write their own custom loggers

### CODE REVIEW

**Comment by alberttorosyan** [comment]
Adding @devfox-se as a reviewer

**Review by devfox-se — APPROVED** [review]
lgtm - also applied a minor fix related to omegaconf

File: aim/sdk/adapters/pytorch_lightning.py [hunk 1]
```diff
@@ -1,129 +1,107 @@
...
                  repo: Optional[str] = None,
                  experiment: Optional[str] = None,
                  train_metric_prefix: Optional[str] = 'train_',
                  val_metric_prefix: Optional[str] = 'val_',
                  test_metric_prefix: Optional[str] = 'test_',
                  system_tracking_interval: Optional[int]
-                 = DEFAULT_SYSTEM_TRACKING_INT,
-                 ):
-        pass
+                 = DEFAULT_SYSTEM_TRACKING_INT,):
```
Comment (alberttorosyan) [inline aim/sdk/adapters/pytorch_lightning.py:28]: worth adding system params auto-tracking flag as well. check the `Run.__init__` for details.

**Review by alberttorosyan — APPROVED** [review]
left one comment.
the rest looks good

File: aim/sdk/adapters/pytorch_lightning.py (hunk 1 above)
Comment (mihran113) [reply aim/sdk/adapters/pytorch_lightning.py:28]: Done ✅

**Pull request merged**